### PR TITLE
Remove crossgen from superpmi tools

### PR DESCRIPTION
**Pull request by kunalspathak**
Now that https://github.com/dotnet/runtime/pull/53458 has excluded `crossgen`, this PR stops collecting superpmi collection for `crossgen` and its references from `superpmi.py`.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -903,21 +902,6 @@ def __init__(self, coreclr_args):
 
         """
 
-        if coreclr_args.host_os == "OSX":
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:906]: Why did you remove these? the collection_shim_name and corerun_tool_name are still used.

**Review by BruceForstall — CHANGES_REQUESTED** [review]

File: src/coreclr/scripts/superpmi.py (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/scripts/superpmi.py:906]: Sorry, that was by mistake. I meant to delete just a single line related  to `crossgen`.

**Review by BruceForstall — APPROVED** [review]

**Comment by kunalspathak** [comment]
Build failures seems to be unrelated.
